packages/extension/src/components/pages/AddNewTrustline/AddNewTrustline.tsx

```diff
@@ -285,77 +252,75 @@ export const AddNewTrustline: FC = () => {
     isParamsMissing: boolean
   ) => {
     const flags = updateFlags(noRipple);
-    setParams({
-      id: params.id,
-      // BaseTransaction fields
-      fee: null,
-      sequence: null,
-      accountTxnID: null,
-      lastLedgerSequence: null,
-      memos: null,
-      signers: null,
-      sourceTag: null,
-      signingPubKey: null,
-      ticketSequence: null,
-      txnSignature: null,
-      // SetTrustline fields
-      limitAmount: {
-        currency: token,
-        issuer: issuer,
-        value: limit
-      },
-      flags: flags,
-      // UI specific fields
-      showForm: showForm,
-      inAppCall: true
-    });
+    const wallet = getCurrentWallet();
+    if (wallet) {
+      setParams({
+        id: params.id,
+        // BaseTransaction fields
+        transaction: buildTrustSet(
+          {
+            ...params.transaction,
+            ...(flags && { flags }),
+            limitAmount: {
+              currency: token,
+              issuer: issuer,
+              value: limit
+            }
+          },
+          wallet
+        ),
+        // UI specific fields
+        showForm: showForm,
+        inAppCall: true
+      });
+    }
 
     setIsParamsMissing(isParamsMissing);
   };
 
   const formInitialValues = useMemo(() => {
-    if (!params.limitAmount) return undefined;
+    if (!params.transaction?.LimitAmount) return undefined;
 
-    const noRipple = params.flags
-      ? typeof params.flags === 'object'
-        ? params.flags.tfSetNoRipple ?? false
-        : !!(params.flags & TrustSetFlagsBitmask.tfSetNoRipple)
+    const noRipple = params.transaction?.Flags
+      ? typeof params.transaction?.Flags === 'object'
+        ? params.transaction.Flags.tfSetNoRipple ?? false
+        : !!(params.transaction.Flags & TrustSetFlagsBitmask.tfSetNoRipple)
       : false;
 
     return {
-      issuer: params.limitAmount.issuer,
-      token: params.limitAmount.currency,
-      limit: Number(params.limitAmount.value),
+      issuer: params.transaction?.LimitAmount.issuer,
+      token: params.transaction?.LimitAmount.currency,
+      limit: Number(params.transaction.LimitAmount.value),
       noRipple
     };
-  }, [params.limitAmount, params.flags]);
+  }, [params.transaction]);
```

Comment:
```suggestion
  }, [params.transaction?.LimitAmount]);
```

Reply:
then it should be
[params.transaction?.Flags, params.transaction?.LimitAmount]